src/top/utop.el

```diff
@@ -645,7 +648,10 @@ it is started."
        (setq utop-completion nil))
       ;; A new possible completion
       ("completion"
-       (push argument utop-completion))
+       (catch 'done
+	 (dolist (prefix utop-completion-prefixes)
+	   (if (string-prefix-p prefix argument)
+	       (progn (push argument utop-completion) (throw 'done t)) nil))))
```

Comment:
Why do you need the `nil` at the end?

Reply:
Sorry, leftover from a previous version which had also an else branche. Fixed.